WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -2832,59 +2832,13 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
-                                data.getStringArrayExtra(MediaPickerConstants.EXTRA_MEDIA_QUEUED_URIS));
-                        int postId = getImmutablePost().getId();
-                        mFeaturedImageHelper.trackFeaturedImageEvent(
-                                FeaturedImageHelper.TrackableEvent.IMAGE_PICKED_POST_SETTINGS,
-                                postId
-                        );
-                        for (Uri mediaUri : uris) {
-                            String mimeType = getContentResolver().getType(mediaUri);
-                            EnqueueFeaturedImageResult queueImageResult = mFeaturedImageHelper
-                                    .queueFeaturedImageForUpload(
-                                            postId, getSite(), mediaUri,
-                                            mimeType
-                                    );
-                            if (queueImageResult == EnqueueFeaturedImageResult.FILE_NOT_FOUND) {
-                                Toast.makeText(
-                                        this,
-                                        R.string.file_not_found, Toast.LENGTH_SHORT
-                                ).show();
-                            } else if (queueImageResult == EnqueueFeaturedImageResult.INVALID_POST_ID) {
-                                Toast.makeText(
-                                        this,
-                                        R.string.error_generic, Toast.LENGTH_SHORT
-                                ).show();
-                            }
-                        }
-                        if (mEditPostSettingsFragment != null) {
-                            mEditPostSettingsFragment.refreshViews();
-                        }
-                    } else if (data.hasExtra(MediaPickerConstants.EXTRA_MEDIA_URIS)) {
-                        List<Uri> uris = convertStringArrayIntoUrisList(
-                                data.getStringArrayExtra(MediaPickerConstants.EXTRA_MEDIA_URIS));
-                        mEditorMedia.addNewMediaItemsToEditorAsync(uris, false);
-                    } else if (data.hasExtra(MediaPickerConstants.EXTRA_SAVED_MEDIA_MODEL_LOCAL_IDS)) {
-                        int[] localIds = data.getIntArrayExtra(MediaPickerConstants.EXTRA_SAVED_MEDIA_MODEL_LOCAL_IDS);
-                        int postId = getImmutablePost().getId();
-                        for (int localId : localIds) {
-                            MediaModel media = mMediaStore.getMediaWithLocalId(localId);
-                            if (media != null) {
-                                mFeaturedImageHelper.queueFeaturedImageForUpload(postId, media);
-                            }
-                        }
-                        if (mEditPostSettingsFragment != null) {
-                            mEditPostSettingsFragment.refreshViews();
-                        }
+                    if (WPMediaUtils.shouldAdvertiseImageOptimization(this)) {
```

Comment:
🚫 Same here. This check is causing a bug where you can only upload the image (in the Post Editor) when the pop-up appears since it doesn't have an else clause simply calling `handlePickerResult`.

```kotlin
if (WPMediaUtils.shouldAdvertiseImageOptimization(this)) {
    WPMediaUtils.advertiseImageOptimization(this, () -> handlePhotoPickerResult(data));
} else {
    handlePhotoPickerResult(data);
}
```

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-Android/pull/19556/commits/29404be05325314ccbeff5623cc423d0106eb6b9, this was a big oversight on my part, thank you for spotting!